### PR TITLE
402/step 3

### LINKED ISSUE #402: Pooling interface: Step 3 - Create strategy
Creates strategy.

It makes the actual call to BatchExchange contract with all the orders.

Details: https://docs.google.com/document/d/1yNddJk6eW1NluEh5B8IL-UghDv9tWnl0ve17SgZBf1k/edit

image

Related to: https://github.com/gnosis/dex-react/issues/400

### PR DESCRIPTION
**Pull request by alfetopito**
Closes #402 

Implemented the interface defined on #402 and also placing the orders.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr435--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr435--dexreact.review.gnosisdev.com

File: src/components/PoolingWidget/index.tsx [hunk 1]
```diff
@@ -128,9 +132,46 @@ function addRemoveMapItem(map: Map<number, TokenDetails>, newToken: TokenDetails
   return copyMap
 }
 
+// TODO: Decide the best place to put this. This file is too long already, but feels to specific for utils
```
Comment (anxolin) [inline src/components/PoolingWidget/index.tsx]: I like the comments below, also agree that the "combinations" should be done in a util

**Comment by alfetopito** [comment]
> Amazing!
> 
> Can we delete the link? Underlined is enough.

Right! I thought it was link to somewhere but didn't know where to point at yet. Will fix.

> image
> 
> Also, I saw that we wait for the TX to be mined until we go to next step.
> In the proposal, I was suggesting to go to next step as soon as the tx is sent, not mined. However, maybe you are right and it's better to wait for it to be mined, just in case we need a re-attempt. In that case, maybe we need to show something like the message of step 4 in step 3:
> image
> 
> What do you think is best?

Hmmm, yeah. I kind of remembered about this step but did not implement it properly.

If we are to show the message of step 4, does it make sense to still have the step 4?

Otherwise, I can fix and move straight to step for before tx is mined.

One issue that I have with this though is, once tx is sent, there's no going back. How is best to translate this into the interface?

> 
> One question. How much gas are you using? Are you estimating somehow?

I'm using Dima's gas estimation

**Pull request merged**